src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -483,6 +483,15 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         // when changing feature sets each lib is separately cached.
         self.resolve.features_sorted(unit.pkg.package_id()).hash(&mut hasher);
```

Comment:
We've always mixed in features for the package itself. Just not for the deps. See this line